### PR TITLE
Fixes issue #13

### LINKED ISSUE #13: Inconsistent / Poor code style
## TL;DR

Code style is currently inconsistent, and some choices are questionable (e.g. functions names).
I think we should stablish a fixed code style to avoid messy code in the future.
Feel free to give suggestions in the comments.

## Detailed explanation

I know code style is something that really depends on the team's / person's preferences, but there are a lot of lines that most people would agree are not part of any code style. 

Here are some lines of code (of many), just to exemplify what I'm trying to say:

<details>
<summary> 
Mix of Spaces and Tabs (not visible on github code preview)
</summary>

mix-space-tab

</details>

<details>
<summary>
Inconsistent use of curly brackets
</summary>

Right after function declaration:
https://github.com/courselab/snaskii21/blob/33eb585ebc9fada532cc6efbb899f2c4dbf250a9/src/ttsnake.c#L292-L294

On the next line:
https://github.com/courselab/snaskii21/blob/33eb585ebc9fada532cc6efbb899f2c4dbf250a9/src/ttsnake.c#L307-L311

</details>

<details>
<summary>
Inconsistent positioning of the * character on pointers
</summary>

https://github.com/courselab/snaskii21/blob/33eb585ebc9fada532cc6efbb899f2c4dbf250a9/src/ttsnake.c#L335-L337
https://github.com/courselab/snaskii21/blob/33eb585ebc9fada532cc6efbb899f2c4dbf250a9/src/ttsnake.c#L359-L361
https://github.com/courselab/snaskii21/blob/33eb585ebc9fada532cc6efbb899f2c4dbf250a9/src/ttsnake.c#L375
https://github.com/courselab/snaskii21/blob/33eb585ebc9fada532cc6efbb899f2c4dbf250a9/src/ttsnake.c#L228
https://github.com/courselab/snaskii21/blob/33eb585ebc9fada532cc6efbb899f2c4dbf250a9/src/ttsnake.c#L123

</details>

## Proposed standard coding style

Based on the majority of lines already written, I've observed the following pattern:

- variables are named in `snake_case`
- functions nouns are not separated at all, so `thislonglongfunctionwouldbesomethinglikethat();`
- constants are `ALL_CAPS_WITH_UNDERSCORE`
- macros are the same as functions (e.g. sysfatal)
- structs are `snake_case` like variables, but its typedef is `snake_case_t` (`_t` at the end)
- **most** of the identation is set to use only SPACES (2 spaces) (not tabs).
- **most** of code blocks' open curly bracket starts on the next line
- pointer asterisk positioning doesn't seem to have any pattern

Here is my opinion on those topics:

As for functions and macros being `everythnigtogheterprettyunreadable`, I really think it's a bad idea to follow this code style, although I see where this came from (ncurses functions use this style). I also think mixing pointer asterisk placement can be confusing for some people, so I would suggest sticking to only one in all scenarios.

All the other stuff I've listed before are pretty standard, but are not consistent throughout the code.

For the pointer placement, my personal preference is to use `type *pointer1, *pointer2;` for declarations of pointers, `void *func(type *pointer);` 

For function names, maybe we should use `snake_case`, just like `pthread_create`.
https://github.com/courselab/snaskii21/blob/33eb585ebc9fada532cc6efbb899f2c4dbf250a9/src/ttsnake.c#L459

Variables can still be `snake_case` as well, but I like to change them to `cammelCase` in my codes to differentiate them (just personal preference).

I think this needs discussion, and I want to emphasize I am not imposing any code style and I'm opening this issue so that we can discuss a code style to stick to throughout the development of this project.

### PR DESCRIPTION
**Pull request by gabriel-libardi**
Fixes #13.

This PR proposes the following changes:
- Fix inconsistent/poor coding style.
- Implement the coding style that was established by the poll (see discussion at #13)

### CODE REVIEW

File: gui/snake.c [hunk 1]
```diff
@@ -1,9 +1,12 @@
...
 #include <stdlib.h>
 
-// Deque functions. Only functions relevant to the game functionality and needs were implemented.
-// Deque has it's head at left end and it's tail at right end. 
-PositionNode* initialize_node(int positionX, int positionY) {
+
+/* Deque functions. Only functions relevant to the 
+   game functionality and needs were implemented.
+   Deque has it's head at left end and it's tail at right end. */
+
```
Comment (jufantozzi) [inline gui/snake.c]: Nit:
remove line breaks between function comment and function code

File: gui/snake.c [hunk 2]
```diff
@@ -82,97 +90,119 @@ void deque_pop_back(Deque* deque) {
...
 
     --deque->size;
     free_node(&node);
 }
 
-// Snake functions
-void initialize_snake(Snake* snake, int headX, int headY, int blockSize) {
+
+/* Snake functions. */
+
```
Comment (jufantozzi) [inline gui/snake.c]: same

**Review by jufantozzi — APPROVED** [review]
Just left some nits, looking good 👌

File: gui/snake.c (hunk 1 above)
Comment (gabriel-libardi) [inline gui/snake.c]: Good point, I missed this when I was reviewing the code. 
```suggestion
```

File: gui/snake.c (hunk 2 above)
Comment (gabriel-libardi) [reply gui/snake.c]:
```suggestion
```

**Review by garibu — APPROVED** [review]
LGTM

**Pull request merged**